### PR TITLE
Fr catches

### LINKED ISSUE #12: Setup destroy fish endpoint

### LINKED ISSUE #13: Setup edit fish endpoint

### PR DESCRIPTION
**Pull request by fredrondina**
### Issue(s) related to this PR:
* closes #
#13 #12 #11 #7 #5 

### Type of change made:
* [ ] Bugfix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
* [ ] Styling- no new features

#### If this submission is a new feature:
* [x] Have you added an explanation of what your changes do?
* [ ] Have you lint your code locally prior to submission?

### Change Log:
*
Added endpoints for creating a new catch, editing a catch, viewing all catches, and removing a catch.

### Were there any challenges that arose? If so, how were the addressed?
* The show endpoint is going to have multiple pieces of functionality, will pair with @tylertomlinson for that 

### Technical Debt:
*Still need to add sad path test

### Testing:
* [X] Have you written new tests for your core changes, as applicable?
* [X] Have you successfully ran tests with your changes locally?
* [X] All new and existing tests passed?
* [ ] Have you run linter?

**Test coverage** = 100%

### CODE REVIEW

File: app/controllers/api/v1/catches_controller.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+class Api::V1::CatchesController < ApplicationController
```
Comment (tylertomlinson) [inline app/controllers/api/v1/catches_controller.rb:1]: Can add  `protect_from_forgery with: :null_session`

File: app/controllers/api/v1/catches_controller.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  def destroy
+    destroyed_catch = Catch.find(params[:id])
+    Catch.delete(params[:id])
+    render json: CatchSerializer.new(destroyed_catch)
+  end
+
+  private
+
+  def catch_params
+    params.require(:catch).permit(:species, :location, :weight, :length, :image, :fly_id)
```
Comment (tylertomlinson) [inline app/controllers/api/v1/catches_controller.rb]: Can also remove the `.require`

File: spec/requests/api/v1/catches_request_spec.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  it "can delete a particular catch by id" do
+    delete "/api/v1/catches/#{@catch4.id}"
+    expect(response).to be_successful
+    expect(Catch.all.count).to eq(3)
+    expect{Catch.find(@catch4.id)}.to raise_error(ActiveRecord::RecordNotFound)
+  end
+
+  it "can create a new catch with attributes and a fly id" do
+    new_catch_params = {species: "New Species", location: "Moms House", length: 111, weight: 45, fly_id: @fly1.id}
+    post "/api/v1/catches", params: {catch: new_catch_params}
```
Comment (tylertomlinson) [inline spec/requests/api/v1/catches_request_spec.rb]: If `.require` is remove on controller can remove the hash and just make `new_cath_params` in parentheses

File: spec/requests/api/v1/catches_request_spec.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  end
+
+  it "can edit a particular catches attributes" do
+    catch_params = {species: "Updated Species", location: "Moms House", length: 222, weight: 01, fly_id: @fly1.id}
+    patch "/api/v1/catches/#{@catch1.id}", params: {catch: catch_params}
+    expect(response).to be_successful
+    updated_catch = Catch.find(@catch1.id)
+    expect(updated_catch.species).to eq("Updated Species")
+  end
+end
```
Comment (tylertomlinson) [inline spec/requests/api/v1/catches_request_spec.rb:47]: nice testing! 💯

**Review by tylertomlinson — CHANGES_REQUESTED** [review]
Let's change those before merging to master

File: app/controllers/api/v1/catches_controller.rb (hunk 1 above)
Comment (fredrondina) [reply app/controllers/api/v1/catches_controller.rb:1]: this has been added, good catch :)

File: spec/requests/api/v1/catches_request_spec.rb (hunk 4 above)
Comment (fredrondina) [reply spec/requests/api/v1/catches_request_spec.rb:47]: haha thanks my dude 👊

File: app/controllers/api/v1/catches_controller.rb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  def destroy
+    destroyed_catch = Catch.find(params[:id])
+    Catch.delete(params[:id])
+    render json: CatchSerializer.new(destroyed_catch)
+  end
+
+  private
+
+  def catch_params
+    params.permit(:species, :location, :weight, :length, :image, :fly_id)
```
Comment (fredrondina) [inline app/controllers/api/v1/catches_controller.rb:25]: removed nested :catch requirement for params

**Review by fredrondina — COMMENTED** [review]
Requested changes have been made please review @tylertomlinson

**Pull request merged**